app/services/foreman_rh_cloud/branch_info.rb

```diff
@@ -0,0 +1,56 @@
+module ForemanRhCloud
+  class BranchInfo
+    def generate(uuid, host, branch_id, request_hostname)
+      {
+        :remote_leaf => uuid,
+        :remote_branch => branch_id,
+        :display_name => host.organization.name,
+        :hostname => request_hostname,
+        :product => {
+          :type => core_app_name,
+          :major_version => core_app_version.major,
+          :minor_version => core_app_version.minor,
+        },
+        :organization_id => host.organization.id,
+        :satellite_instance_id => Foreman.instance_id,
+        :labels => host_labels(host)
+      }
+    end
+
+    def core_app_name
+      return 'Satellite' if defined? ForemanThemeSatellite
```

Comment:
Should this be handled by theme plugin instead?

Reply:
Is it a customer facing string? The theme will brand all translated (and customer facing strings should be translated) strings. If it's saved in the DB or reported somewhere else, by default the theme will not handle this change.
If you still want to brand this string for downstream, we will need to add override code in the theme.